### ISSUE TITLE
[Gateway API v1.2] Implement GEP-3155

### ISSUE THREAD

**Issue by guydc** [envoyproxy/gateway#4381]
*Description*:

Scope: 
- Gateway: BackendTLS
- BackendTLSPolicy: SubjectAltNames

Out-of-Scope:
- BackendTLSPolicy Options : this is implementation-specific, and EG currently uses a typed approach for TLS settings in CTP and EP. Can be left out. 

[optional *Relevant Links*:]
- https://github.com/kubernetes-sigs/gateway-api/pull/3180
- https://gateway-api.sigs.k8s.io/geps/gep-3155

**Comment by alexwo** [comment]
Hi @muwaqar,
Thank you for reaching out. This activity has not started yet,If you’re interested in helping with the implementation you can step in. I will un assign my self from handling this ticket

**Comment by muwaqar** [comment]
@arkodg @guydc what should be the behavior of Gateway `spec.backendTLS.clientCertificateRef` with EG EnvoyProxy `spec.backendTLS.clientCertificateRef` when both are specified?

Should we error out if they both refer to different resources? or have EnvoyProxy field eclipse the Gateway? or just ignore the EnvoyGateway one?

It seems once we have this part of GEP-3155 implemented we can just remove the EnvoyProxy field (not sure of backwards API compatibility policy with EG types).

**Comment by arkodg** [comment]
@muwaqar we prioritize the Gateway API field if both exist

**Comment by arkodg** [comment]
[SAN support](https://gateway-api.sigs.k8s.io/geps/gep-3155/#sans-on-backendtlspolicy) is being added with https://github.com/envoyproxy/gateway/pull/6092

**Comment by arkodg** [comment]
blocked on https://github.com/envoyproxy/gateway/issues/5800#issuecomment-3550412835

**Comment by cnvergence** [comment]
will take a look at this one

**Comment by guydc** [comment]
closing with #8380. cc @zirain
